### ISSUE TITLE
bug: HTTPHandler doesn't respect formatter, only produces logs in key=val format

### ISSUE THREAD

**Issue by BillyUdders** [meltano/meltano#8375]
### Meltano Version

3.2.0

### Python Version

3.11

### Bug scope

CLI (options, error messages, logging, etc.)

### Operating System

Mac OS X, Linux

### Description

Using the below http-logging.yaml with meltano `el` command like so: `meltano --log-config http-logging.yaml el tap-name target-name` does not produce JSON formatted log messages.

```
version: 1
disable_existing_loggers: false
formatters:
  json:
    (): meltano.core.logging.json_formatter
handlers:
  http:
    class: logging.handlers.HTTPHandler
    level: INFO
    formatter: json
    host: 'localhost:3000'
    url: '/log'
    method: 'POST'
root:
  level: DEBUG
  propagate: true
  handlers: [http]
```

When I run the command I get log messages with a body sent to my echo server running that look like this:

```
name=meltano.cli.elt&msg=%7B%27name%27%3A+%27meltano%27%2C+%27run_id%27%3A+%27424571c7-e70d-4473-a5c8-381fed100276%27%2C+%27state_id%27%3A+%272024-01-23T1
60909--tap-n-target-names%27%2C+%27event%27%3A+%27Extract+%26+load+complete%21%27%2C+%27level%27%3A+%27info%27%2C+%27timestamp%27%3A+%272024-01-23T
16%3A09%3A23.403021Z%27%7D&args=%28%29&levelname=INFO&levelno=20&pathname=%my_path_name$24
hon3.11%2Fsite-packages%2Fmeltano%2Fcli%2Felt.py&filename=elt.py&module=elt&exc_info=None&exc_text=None&stack_info=None&lineno=509&funcName=_run_extract_load&created=1706026163.4030552&msecs=403.0&relativeCreated=14806.555271148682&thread=8064847872&threadName=MainThread&processName=MainProcess&process=73440&_logger=%3C_FixedFindCallerLogger+meltano.cli.elt+%28DEBUG%29%3E&_name=info
```

Which isn't JSON but also quite hard to parse and noisy. I've double/triple checked this with Logstash running in a docker container on my local machine, and in a pod in an EKS cluster.

**Comment by edgarrmondragon** [comment]
Hey @BillyUdders, thanks for logging!

I think this is unfortunately expected from Python's builtin `logging.handlers.HTTPHandler`. From the docs[^1]:

<blockquote>
<p>Since preparing a record for sending it to a web server is not
the same as a generic formatting operation, using
<a class="reference internal" href="logging.html#logging.Handler.setFormatter" title="logging.Handler.setFormatter"><code class="xref py py-meth docutils literal notranslate"><span class="pre">setFormatter()</span></code></a> to specify a
<a class="reference internal" href="logging.html#logging.Formatter" title="logging.Formatter"><code class="xref py py-class docutils literal notranslate"><span class="pre">Formatter</span></code></a> for a <a class="reference internal" href="#logging.handlers.HTTPHandler" title="logging.handlers.HTTPHandler"><code class="xref py py-class docutils literal notranslate"><span class="pre">HTTPHandler</span></code></a> has no effect.
Instead of calling <a class="reference internal" href="logging.html#logging.Handler.format" title="logging.Handler.format"><code class="xref py py-meth docutils literal notranslate"><span class="pre">format()</span></code></a>, this handler calls
<a class="reference internal" href="#logging.handlers.HTTPHandler.mapLogRecord" title="logging.handlers.HTTPHandler.mapLogRecord"><code class="xref py py-meth docutils literal notranslate"><span class="pre">mapLogRecord()</span></code></a> and then <a class="reference internal" href="urllib.parse.html#urllib.parse.urlencode" title="urllib.parse.urlencode"><code class="xref py py-func docutils literal notranslate"><span class="pre">urllib.parse.urlencode()</span></code></a> to encode the
dictionary in a form suitable for sending to a web server.</p>
</blockquote>

[^1]: [Official docs for `logging.handlers.HTTPHandler`](https://docs.python.org/3/library/logging.handlers.html#httphandler)

**Comment by edgarrmondragon** [comment]
Can confirm the above:

```python
import logging
from logging.handlers import HTTPHandler
from urllib.parse import urlencode

record = logging.LogRecord("my_logger", logging.INFO, "/path/to/file.py", 10, "EVENT!", (), None)
handler = HTTPHandler("localhost", 5000)
print(urlencode(handler.mapLogRecord(record)))
# name=my_logger&msg=EVENT%21&args=%28%29&levelname=INFO&levelno=20&pathname=%2Fpath%2Fto%2Ffile.py&filename=file.py&module=file&exc_info=None&exc_text=None&stack_info=None&lineno=10&funcName=None&created=1706751680.052684&msecs=52.0&relativeCreated=8.797883987426758&thread=8648734912&threadName=MainThread&processName=MainProcess&process=39301&taskName=None
```
